packages/core/lib/dlc/HyperbolaPayoutCurve.ts

```diff
@@ -0,0 +1,164 @@
+import { PayoutFunctionV1 } from '@node-dlc/messaging';
+import BigNumber from 'bignumber.js';
+
+export class HyperbolaPayoutCurve {
+  constructor(
+    private a: BigNumber,
+    private b: BigNumber,
+    private c: BigNumber,
+    private d: BigNumber,
+    private f_1: BigNumber,
+    private f_2: BigNumber,
+    private positive: boolean = true, // TODO: support negative pieces
+  ) {}
```

Comment:
Add TODO for adding check that ensures that `a * d != b * c`

https://github.com/discreetlogcontracts/dlcspecs/pull/144/files#diff-da39f12c70557dc5785af0351bd22b797660522b55a6fb54c476948ba7c6fcf5R241